### PR TITLE
(Cdap-20) Incremental effort to remove guava from api

### PR DESCRIPTION
**Pull request by chtyim**
There are three changes:

1. Replace usage of TypeToken to Type in FieldAccessor.
2. Change Reflections.visit to take Type instead of TypeToken
3. Remove usage of Guava Joiner and Splitter in cdap-api

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java [hunk 1]
```diff
@@ -42,7 +40,11 @@ private RuntimeArguments() {
   public static Map<String, String> fromPosixArray(String[] args) {
     Map<String, String> kvMap = new HashMap<>();
     for (String arg : args) {
-      kvMap.putAll(Splitter.on("--").omitEmptyStrings().trimResults().withKeyValueSeparator("=").split(arg));
+      int idx = arg.indexOf('=');
+      int keyOff = arg.startsWith("--") ? "--".length() : 0;
+      String key = idx < 0 ? arg.substring(keyOff) : arg.substring(keyOff, idx);
+      String value = idx < 0 ? "" : arg.substring(idx + 1);
```
Comment (gokulavasan) [inline cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java]: what if we have arg = "--validKey=" then arg.substring(idx+1) will be out of bounds

Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java]: Good catch.

File: cdap-common/src/main/java/co/cask/cdap/internal/io/FieldAccessorGenerator.java [hunk 2]
```diff
@@ -126,8 +126,9 @@ private void initializeReflectionField(GeneratorAdapter mg, Field field) {
     mg.visitTryCatchBlock(beginTry, endTry, catchHandle, Type.getInternalName(Exception.class));
     mg.mark(beginTry);
 
-    // Field field = findField(classType, "fieldName")
+    // Field field = Fields.findField(classType, "fieldName")
```
Comment (gokulavasan) [inline cdap-common/src/main/java/co/cask/cdap/internal/io/FieldAccessorGenerator.java]: Fields.findField(TypeToken.of(classType), "fieldName") ?

**Comment by gokulavasan** [comment]
Couple of comments. Rest LGTM :+1:

File: cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java (hunk 1 above)
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java]: Actually it won't. The substring requires the begin index > 0 and <= string.length.

Comment (gokulavasan) [inline cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java]: Ah I see. Ok.

File: cdap-common/src/main/java/co/cask/cdap/internal/io/FieldAccessorGenerator.java (hunk 2 above)
Comment (chtyim) [inline cdap-common/src/main/java/co/cask/cdap/internal/io/FieldAccessorGenerator.java]: Fixed.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT2451-6

**Pull request merged**